src/python/pants/backend/adhoc/target_types.py

```diff
@@ -253,6 +253,22 @@ class AdhocToolOutputRootDirField(StringField):
     )
 
 
+class AdhocToolHashOnlySourcesGlobsField(StringSequenceField):
+    alias: ClassVar[str] = "hash_only_sources_globs"
```

Comment:
Maybe this should be named just `hash_only_sources`? The regular `sources` field can take globs but we don't put "_globs" in its name.

Reply:
I may also want to bikeshed the name further, but not yet...

Reply:
> Maybe this should be named just `hash_only_sources`? The regular `sources` field can take globs but we don't put "_globs" in its name.

Fine by me. I am not wedded to any particular name.

Reply:
What about `unmaterialized_sources`?

Reply:
Or `indirect_sources`?

Reply:
I like `unmaterialized_sources`, although `unmaterialized` might be opaque jargon to many users (i.e. require them to read the docs to make any reasonable guess at what it means).

Just brainstorming some other ideas:

- `extra_invalidation_sources`
- `non_sandbox_sources`
- `workspace_invalidation_sources`
- `workspace_only_sources`

Reply:
I like your suggestions of `workspace_invalidation_sources` and `workspace_only_sources` because they incorporate "workspace" since this feature is intended to be paired with use of `workspace_environment`. Indeed, it points to restricting this feature to only be enabled when the workspace environment is set on the `adhoc_tool` / `shell_command` target. If the user is using a regular environment, then they should be relying on ordinary dependencies to express inputs.

I will modify the PR to go with `workspace_invalidation_sources` for now and error if the field is set for non-workspace environments.

Reply:
For now, I am just going to document it should only be used with `workspace_environment`. No need to error or ignore just yet.